src/Features/Core/SolutionCrawler/WorkCoordinator.GlobalOperationAwareIdleProcessor.cs

```diff
@@ -76,10 +76,10 @@ private void OnGlobalOperationStopped(object sender, GlobalOperationEventArgs e)
                         _globalOperationTask = SpecializedTasks.EmptyTask;
                     }
 
-                    protected async Task GlobalOperationWaitAsync()
+                    protected Task GlobalOperationWaitAsync()
                     {
                         // we wait for global operation if there is anything going on
-                        await _globalOperationTask.ConfigureAwait(false);
+                        return _globalOperationTask;
                     }
```

Comment:
You should remove the comment above the modified statement and rename the method to GetGlobalOperationTask, or better make it a property GlobalOperationTask so the caller knows its fetching a task.